### ISSUE TITLE
Plaza and Pedestrian Zones have broke Realistic Parking Feature

### ISSUE THREAD

**Issue by Bigs2013** [CitiesSkylinesMods/TMPE#1655]
### Steps to reproduce
1.  Create Ped Zone
2.  Zone Ped Zone
3.  Let Game run a bit before a car becomes utterly confused and breaks itself blocking up traffic for miles

### Screenshots?

https://gyazo.com/1139ef27e2209102f1ab6eb7d975f513

**Comment by krzychu124** [comment]
Cars get confused because you've placed parking spots near pedestrian streets that cars are not allowed to use.
Parking AI works like this: 
 - citizen is forced to park his car
 - TMPE is searching for parking spot (either building parking or parking lane)
 - TMPE is trying to find path between current car position and parking spot (here everything fails). 
 DLC didn't break Parking AI. The feature has never handled correctly a scenario where access to parking spot is restricted

**Comment by Bigs2013** [comment]
> Cars get confused because you've placed parking spots near pedestrian streets that cars are not allowed to use. Parking AI works like this:
> 
> * citizen is forced to park his car
> * TMPE is searching for parking spot (either building parking or parking lane)
> * TMPE is trying to find path between current car position and parking spot (here everything fails).
>   DLC didn't break Parking AI. The feature has never handled correctly a scenario where access to parking spot is restricted

Nope, Ill attach save file
https://gyazo.com/b939809a92c94f73339fe422da1ab038

**Comment by krzychu124** [comment]
Yeah, savegame would be useful, either file or share via Steam (guide in the wiki)

**Comment by Bigs2013** [comment]
> Yeah, savegame would be useful, either file or share via Steam (guide in the wiki)

https://gyazo.com/b939809a92c94f73339fe422da1ab038

**Comment by Bigs2013** [comment]
> Yeah, savegame would be useful, either file or share via Steam (guide in the wiki)

Here ya go https://steamcommunity.com/sharedfiles/filedetails/?id=2867763274

**Comment by krzychu124** [comment]
They stuck because of your road settings.
Vehicle is waiting on green segment at the bottom. Wants to park at red segment but on the road side that goes downwards.
It's impossible (most likely path is too long since it forces car to go to the main city far on the right to do a U-turn somewhere).

![image](https://user-images.githubusercontent.com/19638970/192168680-77dc9156-36bd-4e51-b6a7-8c3293677f50.png)

Notice lane connections (cars cannot ignore those). U-turn is not allowed, segment on the right allows only for incoming flow of vehicles so vehicles which exit from the main road are trapped inside.
![image](https://user-images.githubusercontent.com/19638970/192168708-684275e8-f337-400b-8281-d8db622fdddb.png)

Reverse direction of the highway road on the right or remove lane connections and allow U-turn and vehicles will immediately unstuck.

Here another potential problem. Depends on where is the vehicle it may also stuck because of no path (cannot use pedestrian street to access parking)
![image](https://user-images.githubusercontent.com/19638970/192168976-e1741ee4-2976-4640-a43d-0b7bc5b3d002.png)

**Comment by Bigs2013** [comment]
Alright well that explains that but what about in the main City at the T intersection, why are vehicles getting stuck there?

**Comment by Bigs2013** [comment]
Also, why don't they just reroute to a different spot instead of getting stuck?

**Comment by Bigs2013** [comment]
> They stuck because of your road settings. Vehicle is waiting on green segment at the bottom. Wants to park at red segment but on the road side that goes downwards. It's impossible (most likely path is too long since it forces car to go to the main city far on the right to do a U-turn somewhere).
> 
> ![image](https://user-images.githubusercontent.com/19638970/192168680-77dc9156-36bd-4e51-b6a7-8c3293677f50.png)
> 
> Notice lane connections (cars cannot ignore those). U-turn is not allowed, segment on the right allows only for incoming flow of vehicles so vehicles which exit from the main road are trapped inside. ![image](https://user-images.githubusercontent.com/19638970/192168708-684275e8-f337-400b-8281-d8db622fdddb.png)
> 
> Reverse direction of the highway road on the right or remove lane connections and allow U-turn and vehicles will immediately unstuck.
> 
> Here another potential problem. Depends on where is the vehicle it may also stuck because of no path (cannot use pedestrian street to access parking) ![image](https://user-images.githubusercontent.com/19638970/192168976-e1741ee4-2976-4640-a43d-0b7bc5b3d002.png)

Alright I went ahead and did a quick test and allowed U-turns at that intersection but cars still got stuck at those intersection, but if you look at the buildings they are trying to go to they are all in the pedestrian ahead hence my belief its to do with the Ped Areas, also PS I did try removing ALL parking lots just as a quick bug test but that didn't fix the issue

**Comment by krzychu124** [comment]
I found some weird issues in the pathfinding. The reason why vehicles can stuck is pathfinding returns path that contains incorrect lanes (non-vehicle), but only if Parking AI is enabled. I'm working on fix.

**Comment by HenryLoenwind** [comment]
> . I'm working on fix.

BTW: I noticed this issue here, too, but even for cars that want to go to a destination that is nowhere near the pedestrian road.

I started my pedestrian road by changing the last 2 segments of a cul-de-sac to be pedestrian roads. Now cars get stuck when driving by the junction that goes into that road, even when they want to go to a target that is 5 blocks away. They seems to always be on the turn lane that leads in the opposite direction of the pedestrian zone, even though they'd need to go straight to get to their destination.

I just watched the Traffic Routes overlay for a couple of minutes and I noticed several cars driving weird routes. One drove past the left turn into its destination, followed the road for 8 blocks, turned around in a side street, passed by its destination, did a u-turn and then turned left into into its home street. This may be the reason why cars that only need to pass by near the pedestrian street segment get suck?

Which reminds me that I spent most of the evening yesterday nailing down lane permissions to make it impossible for my busses to randomly turn left or right instead of following their route. Interestingly this only happend on the street that passes by the cul-de-sac which ends in pedestrian road segments.

TL;DR: This also affects vehicles that have no reason to go into/through a pedestrian road by just being near it.
